scripts/apps/archive/directives/MarkedItemTitle.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * @module superdesk.apps.archive
+ * @ngdoc directive
+ * @name sdMarkedItemTitle
+ * @requires desks
+ * @requires authoring
+ * @requires highlightServce
+ * @requires $timeout
+ * @description
+ *   This directive is used in authoring-topbar to mark a story as a highlight or
+ *   to mark it for a desk.
+ */
+MarkedItemTitle.$inject = ['desks', '$timeout', 'authoring', 'highlightsService'];
+export function MarkedItemTitle(desks, $timeout, authoring, highlightsService) {
+    return {
+        scope: {
+            item: '=item',
+            markField: '@field'
+        },
+        templateUrl: 'scripts/apps/archive/views/marked_item_title.html',
+        // todo(petr): refactor to use popover-template once angular-bootstrap 0.13 is out
+        link: function(scope, elem) {
+            /*
+             * Toggle 'open' class on dropdown menu element
+             * @param {string} isOpen
+             */
+            scope.toggleClass = function(isOpen) {
+                scope.open = isOpen;
+            };
+
+            scope.marks = scope.item[scope.markField];
+            scope.hasMarkItemPrivilege = authoring.itemActions(scope.item).mark_item_for_desks;
+
+            scope.$on('item:' + scope.markField, ($event, data) => {
```

Comment:
This, and the `if` below show that you are trying to implement something that is in fact two directives. So why not have two? This is very error prone.

I think that instead of using jQuery (which we shouldn't use in an Angular app) event listeners we can make a service to communicate to other parts of the application, instead of clicking around using the `.click()` method. I would not merge this PR in its current state.